exporters/otlp/internal/transform/metric.go

```diff
@@ -234,24 +248,44 @@ func sink(ctx context.Context, in <-chan result) ([]*metricpb.ResourceMetrics, e
 // Record transforms a Record into an OTLP Metric. An ErrUnimplementedAgg
 // error is returned if the Record Aggregator is not supported.
 func Record(r export.Record) (*metricpb.Metric, error) {
-	switch a := r.Aggregation().(type) {
-	case aggregation.MinMaxSumCount:
-		return minMaxSumCount(r, a)
-	case aggregation.Sum:
-		return sum(r, a)
+	agg := r.Aggregation()
+	switch agg.Kind() {
+	case aggregation.MinMaxSumCountKind:
+		return minMaxSumCount(r, agg.(aggregation.MinMaxSumCount))
+
+	case aggregation.SumKind:
+		s, ok := agg.(aggregation.Sum)
+
+		if !ok {
+			return nil, fmt.Errorf("%w: %T", ErrIncompatibleAggregation, agg)
+		}
+		sum, err := s.Sum()
+		if err != nil {
+			return nil, err
+		}
+		return scalar(r, sum, r.StartTime(), r.EndTime())
+
+	case aggregation.LastValueKind:
+		lv, ok := agg.(aggregation.LastValue)
+		if !ok {
```

Comment:
Similar here? Can we add a regression test if this is ever a possibility?

Reply:
Done